patches/server/0986-FOV-Scale-Implementation.patch

```diff
@@ -0,0 +1,86 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moulberry <james.jenour@protonmail.com>
+Date: Tue, 6 Jun 2023 02:08:29 +0800
+Subject: [PATCH] FOV Scale Implementation
+
+Modifies the player's fov by a factor
+The multiplier is affected by the client Accessibility option 'FOV Effects'
+
+This works due to the way vanilla calculates FOV scale as a ratio
+between movementSpeed and walkingSpeed:
+
+fovScale = (movementSpeed / walkingSpeed + 1.0f) / 2.0f
+
+Normally, servers might implement a 'zoom' by changing the movementSpeed
+- this behaviour is well known. However, changing the walkingSpeed
+achieves a similar effect without affecting the speed at which the
+player is able to move.
+
+This patch adds API to do exactly that.
+
+(Note that 'walkingSpeed' on the client has no other use than to
+calculate the FOV)
+
+== AT ==
+public-f net.minecraft.network.protocol.game.ClientboundPlayerAbilitiesPacket walkingSpeed
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 98df2463bf41fc736aa6a2b6ddf89e5abde6eb39..954382a1442d581bb3f13bf8e9e94c29335bb300 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -1877,7 +1877,20 @@ public class ServerPlayer extends Player {
+     @Override
+     public void onUpdateAbilities() {
+         if (this.connection != null) {
+-            this.connection.send(new ClientboundPlayerAbilitiesPacket(this.getAbilities()));
++            // Paper start - use fov scale
++            ClientboundPlayerAbilitiesPacket abilitiesPacket = new ClientboundPlayerAbilitiesPacket(this.getAbilities());
++            float fovScale = this.getBukkitEntity().getFovScale();
++            if (fovScale < -1E24f) {
```

Comment:
Could technically reduce this to something like 
```java
if (Math.abs(fovScale) > 1E24F) {
    abilitiesPacket.walkingSpeed = Math.signum(fovScale * abilitiesPacket.walkingSpeed) * 1E-24F;
}
```
to avoid the duplicate code here.